src/configuration.ts

```diff
@@ -124,8 +131,28 @@ export function loadConfigurationFromPath(configFilePath: string) {
         let fileData = fs.readFileSync(configFilePath, "utf8");
         // remove BOM if present
         fileData = fileData.replace(/^\uFEFF/, "");
-        return JSON.parse(fileData);
+        const configFile: IConfigFile = JSON.parse(fileData);
+        configFile.rulesDirectory = getRulesDirectories(configFile.rulesDirectory, path.dirname(configFilePath));
+        return configFile;
+    }
+}
+
+export function extendConfigFile(config: IConfigFile, baseConfig: IConfigFile): IConfigFile {
+    let combinedConfig: IConfigFile = {};
+
+    const baseRulesDirectory = arrayify(baseConfig.rulesDirectory);
+    const configRulesDirectory = arrayify(config.rulesDirectory);
+    combinedConfig.rulesDirectory = configRulesDirectory.concat(baseRulesDirectory);
+
+    combinedConfig.rules = {};
+    for (const name of Object.keys(objectify(baseConfig.rules))) {
+        combinedConfig.rules[name] = baseConfig.rules[name];
+    }
+    for (const name of Object.keys(objectify(config.rules))) {
+        combinedConfig.rules[name] = config.rules[name];
     }
+
+    return combinedConfig;
```

Comment:
ah, sorry about that re: _.defaults. you're right @JKillian 

@unional those look like minor stylistic preferences... I'm fine with the code layout in this function as it is now. in the future, please leave line-by-line comments rather than rewriting the whole implementation... otherwise it gets pretty hard to review your proposed changes